pypolychord/polychord.py

```diff
@@ -356,3 +355,560 @@ def write(var):
+        'base_dir': 'chains',
+        'file_root': 'test',
+        "grade_dims": [nDims],
+        "nlives": {},
+        "seed": -1,
+    }
+    default_kwargs["grade_frac"] = [1.0]*len(default_kwargs["grade_dims"]),
+    default_kwargs["paramnames"] = [(f"p{i}", f"p_{{{i}}}") for i in range(nDims + default_kwargs("nDerived"))],
+
+
+    if not set(kwargs.keys()) <= set(default_kwargs.keys()):
+        raise TypeError(f"{__name__} got unknown keyword arguments {kwargs.keys() - default_kwargs.keys()}")
+    default_kwargs.update(kwargs)
+    kwargs = default_kwargs
+
+    kwargs["grade_dims"] = [int(d) for d in list(kwargs["grade_dims"])]
+    kwargs["nlives"] = {float(logL):int(nlive) for logL, nlive in kwargs["nlives"].items()}
+
+    try:
+        if rank == 0:
+            os.makedirs(kwargs["base_dir"])
+    except OSError:
+        pass
+        
+    try:
+        if rank == 0:
+            os.makedirs(os.path.join(kwargs["base_dir"], "clusters"))
+    except OSError:
+        pass
+
+    if "cube_samples" in kwargs:
+        make_resume_file_kwargs_interface(loglikelihood, kwargs["prior"], **kwargs)
+        read_resume = kwargs["read_resume"]
+        kwargs["read_resume"] = True
+
+    def wrap_loglikelihood(theta, phi):
+        logL, phi[:] = loglikelihood(theta)
+        return logL
+
+    def wrap_prior(cube, theta):
+        theta[:] = kwargs["prior"](cube)
+
+
+    # Run polychord from module library
+    _pypolychord.run(wrap_loglikelihood,
+        wrap_prior,
+        kwargs["dumper"],
+        nDims,
+        kwargs["nDerived"],
+        kwargs['nlive'],
+        kwargs['num_repeats'],
+        kwargs['nprior'],
+        kwargs['nfail'], 
+        kwargs['do_clustering'],
+        kwargs['feedback'],
+        kwargs['precision_criterion'],
+        kwargs['logzero'],
+        kwargs['max_ndead'],
+        kwargs['boost_posterior'],
+        kwargs['posteriors'],
+        kwargs['equals'],
+        kwargs['cluster_posteriors'],
+        kwargs['write_resume'],
+        kwargs['write_paramnames'],
+        kwargs['read_resume'],
+        kwargs['write_stats'],
+        kwargs['write_live'],
+        kwargs['write_dead'],
+        kwargs['write_prior'],
+        kwargs['maximise'],
+        kwargs['compression_factor'],
+        kwargs['synchronous'],
+        kwargs['base_dir'],
+        kwargs['file_root'],
+        kwargs["grade_frac"],
+        kwargs["grade_dims"],
+        kwargs["nlives"], 
+        kwargs["seed"],
+    )
+
+    if "cube_samples" in kwargs:
+        kwargs["read_resume"] = read_resume
+
+    polychord_output = PolyChordOutput(kwargs["base_dir"], kwargs["file_root"])
+    polychord_output.make_paramnames_files(kwargs["paramnames"])
+    try:
+        import anesthetic
+        return anesthetic.NestedSamples(root=os.path.join(kwargs["base_dir"], kwargs["file_root"]))
+    except ImportError:
+        return polychord_output
+
+
+
+def make_resume_file_kwargs_interface(loglikelihood, **kwargs):
```

Comment:
We should make this a `_make_resume_file` function (to recommend to users not to touch it). You should also re-use the code in one of the functions, rather than have a near direct repetition (a recipe for mistakes in the future).